File: Bronze1-1/06-20-pm/unit02/billboard-I/billboard.cpp
/*
 * billboard.cpp
 */
#include <bits/stdc++.h>
using namespace std;

int main() {
	ifstream cin("billboard.in");
	ofstream cout ("billboard.out");

	int r1[4];
	int r2[4];
	int r3[4];

	//rectangle 1
	for(int i=0; i<4; i++)
		cin >> r1[i];

	//rectangle 2
	for(int i=0; i<4; i++)
		cin >> r2[i];

	//rectangle 3
	for(int i=0; i<4; i++)
		cin >> r3[i];

	//area of overlap of r1 and r3
	int ox1 = max(r1[0], r3[0]);
	int ox2 = min(r1[2], r3[2]);

	int oy1 = max(r1[1], r3[1]);
	int oy2 = min(r1[3], r3[3]);

	int overlap1 = (ox2 - ox1) * (oy2 - oy1);
	if(ox1 >= ox2 || oy1 >= oy2)
		overlap1 = 0;

	//area of overlap of r2 and r3
	int ox21 = max(r2[0], r3[0]);
	int ox22 = min(r2[2], r3[2]);

	int oy21 = max(r2[1], r3[1]);
	int oy22 = min(r2[3], r3[3]);

	int overlap2 = (ox22 - ox21) * (oy22 - oy21);
	if(ox21 >= ox22 || oy21 >= oy22)
		overlap2 = 0;

	int res = (r1[2]-r1[0]) * (r1[3]-r1[1]) + (r2[2]-r2[0]) * (r2[3]-r2[1]);
	res -= (overlap1 + overlap2);

	cout << res << endl;
}

File: Bronze1-1/06-20-pm/unit02/billboard-I/billboard.java
/*
 * Billboard (Dec 2017) 
 * An efficient implementation.
 */
import java.util.*;
import java.io.*;
public class billboard {
	public static void main(String[] args) throws IOException {
		Scanner in = new Scanner(new File("billboard.in"));
		PrintWriter out = new PrintWriter(new File("billboard.out"));
		
		int x1 = in.nextInt();  //billboard 1
		int y1 = in.nextInt();
		int x2 = in.nextInt();
		int y2 = in.nextInt();
		
		int x3 = in.nextInt();  //billboard 2
		int y3 = in.nextInt();
		int x4 = in.nextInt();
		int y4 = in.nextInt();
		
		int x5 = in.nextInt();  //truck
		int y5 = in.nextInt();
		int x6 = in.nextInt();
		int y6 = in.nextInt();
	
		//find the area of overlap between two billboards with the truck
		int overlap1 = Math.max(Math.min(x2, x6)-Math.max(x1, x5), 0) *
				Math.max(Math.min(y2, y6)-Math.max(y1, y5), 0);
		
		int overlap2 = Math.max(Math.min(x4, x6)-Math.max(x3, x5), 0) *
				Math.max(Math.min(y4, y6)-Math.max(y3, y5), 0);
		
		int area1 = (x2-x1) * (y2-y1);
		int area2 = (x4-x3) * (y4-y3);

		out.println( area1 + area2 - overlap1 - overlap2 );
		out.close();
		in.close();
	}
}
